test/errorHandling.js

```diff
@@ -181,140 +181,207 @@ describe('#errorHandling', () => {
       });
 
       if (serverType === 'uds' && clientType === 'client') {
-        it('should re-create the socket on bad connection error for type uds', (done) => {
-          const code = badUDSConnectionCode();
+        describe('#udsSocket', () => {
+
+          // ensure we restore the original `Date.now` after each test
           const realDateNow = Date.now;
-          Date.now = () => '4857394578';
-          // emit an error, like a socket would
-          server = createServer('uds_broken', opts => {
-            const client = statsd = createHotShotsClient(Object.assign(opts, {
-              protocol: 'uds',
-              errorHandler(error) {
-                assert.ok(error);
-                assert.strictEqual(error.code, code);
-              }
-            }), 'client');
-            const initialSocket = client.socket;
-            setTimeout(() => {
-              initialSocket.emit('error', { code });
-              assert.ok(Object.is(initialSocket, client.socket));
-              // it should not create the socket if it breaks too quickly
-              // change time and make another error
-              Date.now = () => 4857394578 + 1000; // 1 second later
-              initialSocket.emit('error', { code });
+          afterEach(() => {
+            Date.now = realDateNow;
+          })
```

Comment:
When trying to confirm that my new test cases break without this PR's change, I found that other timing related tests broke, most likely because `Date.now` wasn't properly restored after an assertion error. Using `afterEach` within a sub-`describe` block ensures that we always restore `Date.now` to the original.